core/src/com/unciv/logic/map/MapUnit.kt

```diff
@@ -850,6 +850,8 @@ class MapUnit {
     }
 
     fun removeFromTile() = currentTile.removeUnit(this)
+    // ONLY FOR DEBUG/Unit Testing
+    fun testingSetCurrentTile(tile: TileInfo) { currentTile = tile }
```

Comment:
I disagree with this existing, There must be a nicer way to do this.

Reply:
Tentative opinion: Unit tests should conform to useful APIs??? So if the test needs to move a unit, use an existing move function... Or if the test just needs a unit with certain characteristics on the new tile, (re-)create one...

Reply:
Understandable. Just means that the tile I use has to be more fully fleshed out in order to use putInTile

Reply:
"fleshed out" - maybe be lazy and use micro-maps loaded using the std map loader [like here](https://github.com/yairm210/Unciv/blob/535c4e9ab0552daa46dca1f0371dc2c773040fb2/tests/src/com/unciv/logic/map/FreshWaterTests.kt). Main caveat: Such a map isn't readily visualizable to the coder. Yes, a TileMap loaded from json _can_ include units, we just do not have the editor for them.

(Idea: Build an Android Studio plugin that actually draws such a map from the string literal...)

Don't take me too seriously here!

Reply:
The real problem is that this check requires the MapUnit to check it's currentTile. So one way to set currentTile is via moveThroughTile. Which also needs the ruleset and to run updateVisibleTiles which needs to know the Current civilization to evaluate. Which means we basically need a full game running.
The other option is via the TileInfo.setUnitTransients(). Except for some reason it keeps complaining I haven't set a ruleset property even though I have. Ugh

Reply:
Quickest way is make currentTile not internal